application/application.go

```
@@ -23,15 +30,43 @@ func New(application, service string) Tags {
 }
 
 // Map with all values
-func (a Tags) Map() map[string]string {
+func (app *Tags) Map() map[string]string {
 	allTags := make(map[string]string)
-	allTags["application"] = a.Application
-	allTags["service"] = a.Service
-	allTags["cluster"] = a.Cluster
-	allTags["shard"] = a.Shard
+	allTags["application"] = app.Application
+	allTags["service"] = app.Service
+	allTags["cluster"] = app.Cluster
+	allTags["shard"] = app.Shard
 
-	for k, v := range a.CustomTags {
+	for k, v := range app.CustomTags {
 		allTags[k] = v
 	}
 	return allTags
 }
+
+// AddCustomTagsFromEnv set additional custom tags from environment variables that match the given regex.
+func (app *Tags) AddCustomTagsFromEnv(regx string) {
+	r, err := regexp.Compile(regx)
+	if err != nil {
+		log.Printf("Error creating custom tags: %v\n", err)
+		return
+	}
+
+	env := os.Environ()
+	for _, envVar := range env {
+		k := strings.Split(envVar, "=")[0]
+		if r.Match([]byte(k)) {
+			v := os.Getenv(k)
+			if len(v) > 0 {
+				app.CustomTags[k] = v
+			}
+		}
+	}
+}
+
+// AddCustomTagFromEnv Set a custom tag from the given environment variable.
+func (app *Tags) AddCustomTagFromEnv(varName, tag string) {
+	v := os.Getenv(varName)
```

Maybe return an error if the env var doesn't exist?